cirq/circuits/qasm_output.py

```diff
@@ -311,7 +311,7 @@ def on_stuck(bad_op):
             should_annotate = decomposed != [main_op]
             if should_annotate:
                 output_line_gap(1)
-                if isinstance(main_op, ops.GateOperation):
+                if main_op.gate is not None:
```

Comment:
This changes behavior if `main_op` is `None` since `isinstance(None, SomeClass)` is `false` but dereferencing `None` will cause an `AttributeError`.

Consider adding a test to ensure/document that this is handled as expected.